packages/guides-restructured-text/src/RestructuredText/Parser/Productions/FieldListRule.php

```diff
@@ -0,0 +1,243 @@
+use phpDocumentor\Guides\Nodes\Metadata\AuthorsNode;
+use phpDocumentor\Guides\Nodes\Metadata\ContactNode;
+use phpDocumentor\Guides\Nodes\Metadata\CopyRightNode;
+use phpDocumentor\Guides\Nodes\Metadata\DateNode;
+use phpDocumentor\Guides\Nodes\Metadata\NoCommentsNode;
+use phpDocumentor\Guides\Nodes\Metadata\NoSearchNode;
+use phpDocumentor\Guides\Nodes\Metadata\OrganizationNode;
+use phpDocumentor\Guides\Nodes\Metadata\OrphanNode;
+use phpDocumentor\Guides\Nodes\Metadata\RevisionNode;
+use phpDocumentor\Guides\Nodes\Metadata\TocDephthNode;
+use phpDocumentor\Guides\Nodes\Metadata\TopicNode;
+use phpDocumentor\Guides\Nodes\Metadata\VersionNode;
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\RestructuredText\Parser\Buffer;
+use phpDocumentor\Guides\RestructuredText\Parser\DocumentParserContext;
+use phpDocumentor\Guides\RestructuredText\Parser\LinesIterator;
+use RuntimeException;
+
+use function mb_strlen;
+use function mb_substr;
+use function preg_match;
+use function strtolower;
+use function trim;
+
+/**
+ * @link https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#definition-lists
+ *
+ * @implements Rule<FieldListNode>
+ */
+final class FieldListRule implements Rule
+{
+    private RuleContainer $productions;
+
+    public function __construct(RuleContainer $productions)
+    {
+        $this->productions = $productions;
+    }
+
+    public function applies(DocumentParserContext $documentParser): bool
+    {
+        return $this->isFieldLine($documentParser->getDocumentIterator()->current());
+    }
+
+    public function apply(DocumentParserContext $documentParserContext, ?CompoundNode $on = null): ?Node
+    {
+        $iterator = $documentParserContext->getDocumentIterator();
+        $fieldListItemNodes = [];
+        while ($iterator->valid() && $this->isFieldLine($iterator->current())) {
+            $fieldListItemNodes[] = $this->createListItem($documentParserContext);
+            $iterator->next();
+        }
+
+        if ($on instanceof DocumentNode && !$on->isTitleFound()) {
+            // A field list found before the first title node is considered file wide meta data:
+            // https://www.sphinx-doc.org/en/master/usage/restructuredtext/field-lists.html#file-wide-metadata
+            // It is not output
+            $this->addMetadata($on, $fieldListItemNodes);
+
+            return null;
+        }
+
+        return new FieldListNode(...$fieldListItemNodes);
+    }
+
+    /**
+     * @param FieldListItemNode[] $fieldListItemNodes
+     */
+    private function addMetadata(DocumentNode $documentNode, array $fieldListItemNodes): void
+    {
+        foreach ($fieldListItemNodes as $fieldListItemNode) {
+            switch (strtolower(trim($fieldListItemNode->getTerm()))) {
+                // Bibliographic Fields Defined by docutils
+                // https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#field-lists
+                case 'abstract':
```

Comment:
It would be nice to refactor this into a list of factories that we can configure via the constructor. This will allow us to extend this switch without changing this class. 

As sphinx has a different list of supported fields that docutils has. Other standards could have more types?

Reply:
Should this be factories or productions?